lib/Config.js

```diff
@@ -71,19 +71,78 @@ class Config {
          * It can be overridden using the S3_CONFIG_FILE environment var.
          */
         this._basePath = path.join(__dirname, '..');
-        this.path = path.join(__dirname, '../config.json');
+        this.configPath = path.join(__dirname, '../config.json');
         if (process.env.S3_CONFIG_FILE !== undefined) {
-            this.path = process.env.S3_CONFIG_FILE;
+            this.configPath = process.env.S3_CONFIG_FILE;
+        }
+        this.locationConfigPath = path.join(__dirname,
+          '../locationConfig.json');
+        if (process.env.S3_LOCATION_FILE !== undefined) {
+            this.locationConfigPath = process.env.S3_LOCATION_FILE;
         }
 
         // Read config automatically
+        this._getLocationConfig();
         this._getConfig();
     }
 
+    _getLocationConfig() {
+        let locationConfig;
+        // check exist file
+        try {
+            fs.statSync(this.locationConfigPath);
+        } catch (err) {
+            if (err) {
+                // location file does not exist
+                // old config
```

Comment:
instead of 'old config', I would say 'legacy config format where only one config file'